pinot-core/src/main/java/org/apache/pinot/core/operator/timeseries/TimeSeriesAggregationOperator.java

```diff
@@ -82,55 +82,54 @@ public TimeSeriesAggregationOperator(
 
   @Override
   protected TimeSeriesResultsBlock getNextBlock() {
-    ValueBlock transformBlock = _projectOperator.nextBlock();
-    if (transformBlock == null) {
-      TimeSeriesBuilderBlock builderBlock = new TimeSeriesBuilderBlock(_timeBuckets, new HashMap<>());
-      return new TimeSeriesResultsBlock(builderBlock);
-    }
-    BlockValSet blockValSet = transformBlock.getBlockValueSet(_timeColumn);
-    long[] timeValues = blockValSet.getLongValuesSV();
-    if (_timeOffset != null && _timeOffset != 0L) {
-      timeValues = applyTimeshift(_timeOffset, timeValues);
-    }
-    int[] timeValueIndexes = getTimeValueIndex(timeValues, _storedTimeUnit);
-    Object[][] tagValues = new Object[_groupByExpressions.size()][];
+    ValueBlock valueBlock;
     Map<Long, BaseTimeSeriesBuilder> seriesBuilderMap = new HashMap<>(1024);
```

Comment:
Should we increase the limit of 1k series Builder Map?

Reply:
this is the initial capacity and the map may grow beyond this based on the default lf (0.75 by default).